### PR TITLE
Improve error message for deprecated core expressions

### PR DESCRIPTION
**Pull request by kwin**
Compare with discussion at https://the-asf.slack.com/archives/C7Q9JB404/p1681245267656379

### CODE REVIEW

File: maven-core/src/main/java/org/apache/maven/plugin/internal/DeprecatedCoreExpressionValidator.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ class DeprecatedCoreExpressionValidator extends AbstractMavenPluginParametersVal
 
     @Override
     protected String getParameterLogReason(Parameter parameter) {
-        return "is deprecated core expression; " + DEPRECATED_CORE_PARAMETERS.get(parameter.getDefaultValue());
+        return "is leveraging a deprecated core expression: " + DEPRECATED_CORE_PARAMETERS.get(parameter.getDefaultValue());
```
Comment (kwin) [inline maven-core/src/main/java/org/apache/maven/plugin/internal/DeprecatedCoreExpressionValidator.java:54]: Probably we should even log the used expression.

**Comment by kwin** [comment]
I see this is partially fixed already in https://github.com/apache/maven/pull/1079/files#diff-c8dae894a4075bd2b8071c01ebefbf6c809009809929df69125b98fc3b3c6c17.

**Comment by kwin** [comment]
Superseded by #1079

**Comment by slawekjaranowski** [comment]
source branch deleted

**Pull request closed without merging**